plugins/woocommerce-admin/client/marketing/hooks/useNewCampaignTypes.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY } from '~/marketing/data-multichannel/constants';
+import {
+	CampaignTypesState,
+	CampaignType as APICampaignType,
+	ApiFetchError,
+} from '~/marketing/data-multichannel/types';
+import { CampaignType } from '~/marketing/types/CampaignType';
+
+type UseNewCampaignTypes = {
+	loading: boolean;
+	data?: Array< CampaignType >;
+	error?: ApiFetchError;
+};
+
+export const useNewCampaignTypes = (): UseNewCampaignTypes => {
+	const convert = ( campaignType: APICampaignType ): CampaignType => {
+		return {
+			id: campaignType.id,
+			icon: campaignType.icon_url,
+			name: campaignType.name,
+			description: campaignType.description,
+			createUrl: campaignType.create_url,
+			channelName: campaignType.channel.name,
+			channelSlug: campaignType.channel.slug,
+		};
+	};
+
+	return useSelect( ( select ) => {
+		const { hasFinishedResolution, getCampaignTypes } = select( STORE_KEY );
+		const campaignTypesState = getCampaignTypes< CampaignTypesState >();
+
+		return {
+			loading: ! hasFinishedResolution( 'getCampaignTypes' ),
```

Comment:
Precisely, `! hasFinishedResolution` is not exactly equivalent to the loading status, although it works correctly here but is not very precise in the semantic meaning.

Reply:
> is not very precise in the semantic meaning.

Do you have a suggestion? I'm guessing something like the following?

```suggestion
			hasFinishedResolution: hasFinishedResolution( 'getCampaignTypes' ),
```

And then replace the usage of `loading` with `! hasFinishedResolution`?

Reply:
Maybe `hasResolved`? There are some namings that are the same or start with it in the Gutenberg repo, and also a few in this repo. Another pattern like `fooResolved` might be okay as well.

Reply:
Hmm, but would `hasResolved` be confused with `hasFinishedResolution` vs `hasStartedResolution`? 😂 

We can continue discussing about this naming. If we want to change it, I shall do it in a separate PR, to change all the `loading` to a name that we agree with. For now, I'll leave it as is.

Reply:
> but would `hasResolved` be confused with `hasFinishedResolution` vs `hasStartedResolution`?

`hasStartedResolution` is hardly used, and the semantics of *Resolved* is closer to *Finished* than *Started*, so it shouldn't be that easy to confuse. If this is still a concern, then there is a case for using `hasFinished` and `hasStarted` to distinguish between them.

[The exact same name `hasFinishedResolution`](https://github.com/woocommerce/woocommerce/pull/37044#discussion_r1131043294) is not recommended, although it can be recognized by TS to be a boolean rather than a wp-data `hasFinishedResolution` selector, it is still relatively confusing when viewed on GH.

Reply:
> `hasStartedResolution` is hardly used, and the semantics of _Resolved_ is closer to _Finished_ than _Started_, so it shouldn't be that easy to confuse. If this is still a concern, then there is a case for using `hasFinished` and `hasStarted` to distinguish between them.

Thanks for the ideas! I'll think about this a bit more and I can come up with a separate PR to change them later.